internal/auth/internal/controllers/register.go

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package controllers
+
+import (
+	"github.com/hashicorp/consul/internal/auth/internal/controllers/trafficpermissions"
+	"github.com/hashicorp/consul/internal/controller"
+)
+
+type Dependencies struct {
+	WorkloadIdentityMapper trafficpermissions.TrafficPermissionsMapper
```

Comment:
Naming convention question: Generally, I see the package name omitted from the interface name since it can be a bit redundant otherwise. I notice that we do this same thing in other controllers as well. Do we have a reason for the preference? 

Effective go generally recommends [the other way](https://go.dev/doc/effective_go#interface-names:~:text=Similarly%2C%20the%20function,choose%20good%20names.).

```
trafficpermissions.TrafficPermissionsMapper
vs.
trafficpermissions.Mapper
```